src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -35,15 +38,25 @@ public EnvelopeProcessor(
         this.processEventRepository = processEventRepository;
     }
 
-    public Envelope processEnvelope(byte[] metadataStream, String containerName) throws IOException {
+    public Envelope parseEnvelope(byte[] metadataStream) throws IOException {
         if (Objects.isNull(metadataStream)) {
             throw new MetadataNotFoundException("No metadata file found in the zip file");
         }
         //TODO Perform json schema validation for the metadata file
         InputStream inputStream = new ByteArrayInputStream(metadataStream);
-        Envelope envelope = EntityParser.parseEnvelopeMetadata(inputStream);
-        envelope.setContainer(containerName);
 
+        return EntityParser.parseEnvelopeMetadata(inputStream);
+    }
+
+    public Optional<Envelope> hasEnvelopeFailedToUploadBefore(Envelope envelope) {
```

Comment:
Can't all this be done in sql/jpaql? That could be (much) more efficient depending on the dataset